assets/src/domain/eventEditor/services/apollo/queries/priceTypes/usePriceTypeForPrice.ts

```diff
@@ -21,8 +22,11 @@ const usePriceTypeForPrice = (priceId: EntityId): PriceType => {
 
 	// get the default price type object
 	const defaultPriceType = useDefaultPriceType();
-	let relatedPriceTypes = usePriceTypes(relatedPriceTypeIds);
-	relatedPriceTypes = relatedPriceTypeIds.length ? relatedPriceTypes : [];
+	const allPriceTypes = usePriceTypes();
+
+	const relatedPriceTypes = relatedPriceTypeIds.length
+		? entitiesWithGuIdInArray(allPriceTypes, relatedPriceTypeIds)
+		: [];
```

Comment:
maybe we can move this check and leave this clean?
```
 const entitiesWithGuIdInArray = <T extends Entity>(entities: T[], guidArray: EntityId[]): T[] => {
    if(guidArray.length === 0) return [];
	return filter((entity: T) => includes(entityGuId(entity), guidArray), entities);
};

const relatedPriceTypes = entitiesWithGuIdInArray(allPriceTypes, relatedPriceTypeIds);
```
?

Reply:
Done